pyvrp/read.py

```diff
@@ -552,3 +552,40 @@ def _allowed2profile(self) -> dict[tuple[int, ...], int]:
                 allowed_clients2profile_idx[clients] = next(profile_idx)
 
         return allowed_clients2profile_idx
+
+
+def _problem_data_from_json(cls, data: dict) -> ProblemData:
+    """
+    Creates a :class:`~pyvrp._pyvrp.ProblemData` instance from a JSON-like
+    dictionary.
+
+    Parameters
+    ----------
+    data : dict
```

Comment:
```suggestion
    data
```
(argument types are mostly unnecessary in the docstrings because we already type them in the function signature)